### DISCUSSION TITLE
getting started from zero?

### DISCUSSION THREAD

**Discussion by m9tz** [zalo/CascadeStudio#71] [Q&A]
Hi, I'am not verry used to *.js: so how to start the cascadestudio locally?

for example on a debian testing I start with

1. sudo apt install npm
2. (or/and) sudo npm install -g npm@7.5.2
3. npm install https://github.com/zalo/CascadeStudio
4. ? npm install live-server
5. ?? cd ~/node_modules/opencascade.js
6. ??? npm run index.js 
7. ???? or build an index.html

please bootme up ..

**Comment by owennewo** [comment]
It is much easier than that! 
I'm on Ubuntu using chrome and just installed it as a web app.  

You can use it through the browser
https://zalo.github.io/CascadeStudio/

and then install it locally like this:

https://github.com/zalo/CascadeStudio/blob/master/icon/Installable.gif

**Comment by m9tz** [comment]
ok I can install a chrome app nice! but I can't save or load a file? does the app needs a network-connection? And how does it work in firefox and how offline?

**Reply by owennewo** [reply]
I can try to help but I'm no expert.  I think the CascadeStudio is a 'Progressive Web App' (PWA).  Chrome has good support for this.  Firefox has recently said it is dropping support for PWA - so it may not be possible to get it working.

If you want to get it working properly in chrome I'd go back to: 
https://zalo.github.io/CascadeStudio/
And try to save a file there.  This should prompt you for to allow 'file editing' permission.  I suspect if you do that then that will also fix it in your PWA.

If you are not getting prompted then click on the padlock left of url and click site settings
![image](https://user-images.githubusercontent.com/15650427/107149398-47edd080-6950-11eb-95df-31cfd8d956ef.png)
Scroll down to file editing and check it is not set to 'block'

![image](https://user-images.githubusercontent.com/15650427/107149186-25a78300-694f-11eb-9f86-71140f495af9.png)

I wouldn't expect it to need a network connection after PWA is installed

**Reply by Irev-Dev** [reply]
There's a problem with saving files in browsers apart from chrome and edge that was introduced in this PR.
https://github.com/zalo/CascadeStudio/pull/39
So for the time being you'll have to use chrome to save files.

**Reply by m9tz** [reply]
what do I have to do? I can't save anything nothing hapend when I press any save or load button. Do I have to clone the project to my account? I have install it as an app in chrome under debian/testing. Save files is showing "ask".

**Reply by owennewo** [reply]
How about you forget downloading it as an app for the time being.  Can you go back to https://zalo.github.io/CascadeStudio/ and see if saves using the 'website app'.  I suspect the downloaded app (which has no chrome menu) is not showing you the prompts for some reason.  If the website works then retry the downloaded app.

**Comment by raydeleu** [comment]
I have described the installation procedure at the following location: 

https://github.com/raydeleu/CascadeStudioManual/blob/main/CCS_Manual_main.adoc#local-installation

Here is the text of that section: 

2.2. Local installation

2.2.1. Using a local web server
As the author has published CascadeStudio as an Open Source project, it is possible to download the complete source code from the github page mentioned above. Using the source code it is possible to install a local version on a webserver. Running the program "is as simple as running a server from the root directory (such as the VS Code Live Server, Python live-server, or Node live-server".

The approach with the VS Code live server is indeed very simple. Follow these steps:
1) install VS Code from [https://code.visualstudio.com]
2) Open VSCode and type ctrl+P, type "ext install ritwickdey.liveserver". Alternatively you can open the extension sidebar which opens the Marketplace. If you enter "live server" a long list of extensions is shown. The server from ritwickdey will occur on top of the list as this is by far the most downloaded version.
3) download the code of CascadeStudio from https://github.com/zalo/CascadeStudio by pressing the green "Code" button. Choose "Download ZIP". After downloading unpack the zip file somewhere in your file system.
4) In VS Code, go to "File" and choose the command "Add folder to workspace". Choose the folder "CascadeStudio-master" that you probably just created by unpacking the git repository.
5) Right-click on the file "index.html" and choose "Open with Live Server". In my case my standard browser opened the page "http://127.0.0.1:5500/index.html" and showed the interface to CascadeStudio.

The server seems to run really inside VS Code, so if you quit VS Code the local version of CascadeStudio will also be shut down.

2.2.2. Install CascadeStudio as a Progressive Web App

An even simpler approach is to install CascadeStudio as a Progressive Web App (PWA). A PWA is a local - almost native - application that can run even without an internet connection. This is achieved by installing a socalled "service worker" that continues to provide the functionality of a web application by using a local cache. To the user the PWA looks identical to a normal application that is installed on the computer. It can be installed in the applications folder and the icon can be shown on the desktop and task bar (or dock).

To install CascadeStudio as a Web App perform the following steps:

1) Open the page https://zalo.github.io/cascadestudio/
2) In the browser address bar, click on the "+" sign
3) In the dialog "Install App?" choose "Install"

**Reply by raydeleu** [reply]
The link was changed to https://github.com/raydeleu/CascadeStudioManual.
